system/lib/wasmfs/js_api.cpp

```diff
@@ -105,4 +105,21 @@ void _wasmfs_symlink(char* old_path, char* new_path) {
 long _wasmfs_chmod(char* path, mode_t mode) {
   return __syscall_chmod((intptr_t)path, mode);
 }
+
+// Helper method that identifies what a path is:
+//   ENOENT - if nothing exists there
+//   EISDIR - if it is a directory
+//   EEXIST - if it is a normal file
```

Comment:
It seems odd to use errno here for the positive cases.

I guess its ok?   Maybe not worth defining a new enum?

Reply:
Yeah, maybe it's a little odd, but it saves defining a new enum etc. And this is an internal API, so I tried to keep it simple.

Reply:
No, it can fall back using `readAsync` to load directly without FS support.

Reply:
^ looks like this response was meant for the other discussion thread